files/routes/volunteer_janitor.py

```diff
@@ -0,0 +1,92 @@
+
+from files.__main__ import app
+from files.classes.comment import Comment
+from files.classes.flags import CommentFlag
+from files.classes.user import User
+from files.classes.volunteer_janitor import VolunteerJanitorRecord, VolunteerJanitorResult
+from files.routes.volunteer_common import VolunteerDuty
+from flask import g
+import pprint
+import random
+import sqlalchemy
+
+class VolunteerDutyJanitor(VolunteerDuty):
+
+    def __init__(self, choices):
+        self.choices = choices
+
+    def accept(self, v) -> None:
+        for item in self.choices:
+            record = VolunteerJanitorRecord()
+            record.user_id = v.id
+            record.comment_id = item
+            record.recorded_utc = sqlalchemy.func.now()
+            record.result = VolunteerJanitorResult.Pending
+            g.db.add(record)
+        
+        g.db.commit()
+
+    def embed_template(self) -> str:
+        return "volunteer_janitor.html"
+    
+    def comments(self) -> list[Comment]:
+        return g.db.query(Comment).where(Comment.id.in_(self.choices))
+
+
+def get_duty(u: User) -> VolunteerDutyJanitor:
+    if not app.config['VOLUNTEER_JANITOR_ENABLE']:
+        return None
+
+    # these could probably be combined into one query somehow
+
+    # find reported comments not made by the current user
+    reported_comments = g.db.query(Comment) \
+        .where(Comment.filter_state == 'reported') \
+        .where(Comment.author_id != u.id) \
+        .with_entities(Comment.id)
+
+    reported_ids = [reported.id for reported in reported_comments]
+    
+    if not app.config['DBG_VOLUNTEER_PERMISSIVE']:
+        # find distinguished children of those reported comments
+        distinguished_children = g.db.query(Comment) \
+            .where(Comment.parent_comment_id.in_(reported_ids)) \
+            .where(Comment.distinguish_level > 0) \
+            .with_entities(Comment.parent_comment_id)
```

Comment:
one other consideration: this is... not the best thing to have to do. what about setting a flag on distinguish on the parent comment that marks it as having been seen by a mop. this could also be set on approval or (i think removal?)

Reply:
It's pretty gnarly yeah.

I think this is not going to be a significant perf hit at the moment given the other issues we have; I also think that there's a lot of refactoring that needs to happen anyway (see #359). I'm gonna add a comment to this but punt on it for now.